src/Workspaces/Core/Portable/Workspace/Solution/DocumentInfo.cs

```diff
@@ -73,6 +73,12 @@ internal DocumentInfo(DocumentAttributes attributes, TextLoader? loader, IDocume
             DocumentServiceProvider = documentServiceProvider;
         }
 
+        public static DocumentInfo Create(
+            Document document)
+        {
+            return new DocumentInfo(document.State.Attributes, null, document.State.Services);
```

Comment:
1. that's certainly surprising... how does the ID/name/etc. get passed along?
2. named param for null.

Reply:
Attributes contains the id/name/etc. I took this from another place in code (I should go replace the usage there with this tbh...) and I _think_ it's the best way to duplicate info from a document. @jasonmalinowski  may have some good feedback here

Reply:
Removed and changed to explicit API for each property.